### ISSUE TITLE
Implement Struct (parser part)

### ISSUE THREAD

**Issue by tot0p** [Eclalang/Ecla#55]
Syntax of the struct is : 

```ecla
struct Point {
  x : int ;
  y : int ;
}
```

**Comment by mkarten** [comment]
Here is some explaining about the structs and the selectors

the first changes is that the MethodCallExpr Node does no longer exist

when you write :
```ecla
import "console";
console.println("Hello World ! ");
```
you dont get a Method call but a selector

This is a selector
```ecla
type SelectorExpr struct {
	Field lexer.Token
	Expr  Expr
	Sel   Expr
}
```
it represent any expression that uses a comma between elements like : field access and  library function access

for exemple here is the representation of the code we wrote earlier

![image](https://github.com/Eclalang/Ecla/assets/77793053/699cdfdc-423c-4aa3-a5cd-118c85faa1e8)

whe have Expr equal to console wich is a Var literal and Sel equal to a functionCallExpression of the function println with all the specified parameters

the same case apply i you acces a field in a struct

here is an ecla struct
```ecla
struct Point {
    x : int;
    y : int;
}
```

this struct has 2 fields x and y, if i want to access them i will do it like this
```ecla
#just imagine we have created a struct instance before
console.println(p.x);
console.println(p.y);
```

here is the selector representation : 
![image](https://github.com/Eclalang/Ecla/assets/77793053/6a4177ec-e7dd-405e-80d2-67554b989779)
we have the first selector for the library call to console and in parameter we have another selector that represent p.x
where Expr is equal to p, a Var literal and Sel equal to x, a Var literal too

selectors can also be embeded in each other

if there is a struct inside our Point struct lets name it test

```ecla
struct Test {
    value : string;
}

struct Point {
    x : int;
    y : int;
    test : Test;
}
```

is we access the parameters like so :
```
console.println(p.test.value);
```

we will skip the console.println selector, in the parameters of the function we have a selector with Expr equal to p and Sel equal to another selector with Expr equal to test and Sel equal to value

**Comment by mkarten** [comment]
closed by #54
